handler/handler.go

```diff
@@ -118,6 +132,30 @@ func (h *InstanceComplete) Handle(ctx context.Context, workerID int, msg kafka.M
 	return nil
 }
 
+func (h *InstanceComplete) getFilterInfo(ctx context.Context, filterID string, logData log.Data) ([]string, string, error) {
+	dimensions, _, err := h.filters.GetDimensions(ctx, "", h.cfg.ServiceAuthToken, "", filterID, nil)
+	if err != nil {
+		return nil, "", &Error{
+			err:     fmt.Errorf("failed to get dimensions: %w", err),
+			logData: logData,
+		}
+	}
+
+	dimensionNames := make([]string, 0)
+	for _, d := range dimensions.Items {
+		dimensionNames = append(dimensionNames, d.Name)
+	}
+
+	model, _, _ := h.filters.GetJobState(ctx, "", h.cfg.ServiceAuthToken, "", "", filterID)
```

Comment:
Couldn't we get the dimension names from this call rather than making the separate calls before?